### ISSUE TITLE
Annoying notification "SFTP: Please download locations selected"

### ISSUE THREAD

**Issue by rugk** [andyholmes/gnome-shell-extension-gsconnect#882]
**Describe the bug**
I always see an annoying notification.

**Steps To Reproduce:**
Just connect a device?

Happens:
>  "[device]: SFTP: Please download locations selected"

![](https://user-images.githubusercontent.com/11966684/85233411-3332eb80-b406-11ea-8caf-84bbb3925787.png)

**Expected behavior**
Dunno?
I guess this is an UX problem. I don't know what this notification is supposed to tell me.
When I click on it, nothing happens.

It keeps reappearing. Seemingly randomly.

I also checked the settings of the device in GSConnect and there is a (default) download location set. The "Downloads" folder.
ANd I also can receive files IIRC, correctly.
So I don't know at all, what this is about…

 - **Paired Device(s):**  Android 9

**Additional Notes:**

Fedora Silverblue 32
GNOME 3.36

**Comment by andyholmes** [comment]
Sounds like a mistranslation in the Android app. This should read something like "No storage locations configured".

This means you are trying to mount the remote device, without specifying a folder in the Android app.

**Comment by rugk** [comment]
Nah, I've tried to translate it from Gernan to English, that's why I had this problem.

But yeah, if that's the case, then it's still an UX problem. Who expects that this is a problem in **the app**! ……… that is suprising…

**Comment by andyholmes** [comment]
Those messages come direct from the Android app, so it would probably have to be amended there to hint that the phone is the problem. The error messages just come as pre-translated strings, so there's not much we can do but show them to the user :man_shrugging:

**Comment by stuartlangridge** [comment]
Clarifying this (because I get the same message), you say "This means you are trying to mount the remote device, without specifying a folder in the Android app", which I'm sure is true, but I don't think I am trying to mount the remote device; I certainly don't intend to do so, and I don't think I turned that on manually, which suggests that this feature defaults to being on. Is there some way to turn off mounting the device? And should it be turned off until it's explicitly turned on? (Perhaps there's an argument that useful features should be available by default so they can be discovered, but this doesn't work out of the box because it also needs configuring in the Android app, which isn't done by default._

**Comment by rugk** [comment]
Yeah, also dunno what setting in the Android app that is. It keeps reappearing, …

The UX thing for GSConnect I can suggest is to amend it with a message like:
> Message from phone: […]

or, depending on what you know:
> Error message from phone XY: […]

**Comment by andyholmes** [comment]
> Is there some way to turn off mounting the device? And should it be turned off until it's explicitly turned on?

You can disable the SFTP plugin in the advanced page of GSConnect, or you can disable it in the Android app (or both). The default behaviour is to request the SFTP server info when a device connects, and if it is received attempt mount it.

The circumstances for why we auto-mount are a little bit convoluted, but its likely your device is disconnecting and reconnecting more often than you think it is, while the SFTP connection often has its own lifetime, and what a "connection session" is in KDE Connect is not well-defined.

**TL;DR** I'd recommend disabling the plugin if you just don't use it, or opening a new issue with a request for that option so we can track the required changes as a group.

> Yeah, also dunno what setting in the Android app that is. It keeps reappearing, …

It's the only related setting in the Android app :)

See *Plugin settings -> Filesystem expose -> Add storage location*

> The UX thing for GSConnect I can suggest is to amend it with a message like...

Sure, we can do that.
